keras/losses.py

```diff
@@ -1959,6 +1960,15 @@ def categorical_crossentropy(
     y_true = tf.cast(y_true, y_pred.dtype)
     label_smoothing = tf.convert_to_tensor(label_smoothing, dtype=y_pred.dtype)
 
+    if y_pred.shape[-1] == 1:
+        warnings.warn(
+            "Recieved an one-dimensional output. "
+            "Consider using binary crossentropy "
+            "instead of categorical crossentropy "
+            "if you have only 2 labels",
```

Comment:
Usually we do the strings in the following format:
```py
f"Expected something to be something. Received: {something}."
```
So my suggestion would be:
```py
"Expected the tensor's shape passed to 'categorical_crossentropy' to be (batch_size, n_classes), "
f"where n_classes > 1. Received: y_pred.shape={y_pred.shape}. "
"Consider using 'binary_crossentropy' if you only have 2 classes."
```

See if @fchollet have comments on this part.